proxy/envoy/config.go

```diff
@@ -167,8 +167,17 @@ func buildHTTPListener(routeConfig *HTTPRouteConfig, ip string, port int, rds bo
 		config.RouteConfig = routeConfig
 	}
 
+	var sslContext *SSLContext
+	// Build the sslContext for http inbound listener (only inbound listener does not use rds)
+	if !rds {
+		sslContext = buildListenerSSLContext(mesh)
```

Comment:
The logic here is a little confusing to me, particularly since `buildListenerSSLContext()` can potentially returns nil. Can we do something like this:
```
if !rds && mesh.authEnabled() {
 sslContext = buildListenerSSLContext()
}
```